python/pyspark/pandas/indexes/base.py

```diff
@@ -2235,6 +2235,24 @@ def union(
         """
         Form the union of two Index objects.
 
+        .. note:: For duplicated values, pandas chooses the number of duplicates of self or other
+            with more duplicates. But counting all duplicates is very expensive for large data,
+            so pandas-on-Spark always chooses the number of duplicates in self.
```

Comment:
Is there any release note in pandas?

Reply:
They mentioned it's bug fix in their release note at https://pandas.pydata.org/pandas-docs/dev/whatsnew/v1.3.0.html#indexing.

Let me try the solution @ueshin commented https://github.com/apache/spark/pull/33634#discussion_r682859453